darshan-util/darshan-logutils.c

```diff
@@ -1175,6 +1175,20 @@ static int darshan_log_get_header(darshan_fd fd)
         fd->mod_ver[DARSHAN_H5D_MOD] = 0;
     }
 
+    if((log_ver_maj == 3) && (log_ver_min < 41))
+    {
+        /* perform module index shift to account for PNETCDF_VAR module from 3.4.1 */
+        memmove(&fd->mod_map[DARSHAN_PNETCDF_VAR_MOD+1],
+            &fd->mod_map[DARSHAN_PNETCDF_VAR_MOD],
+            (DARSHAN_MAX_MODS-DARSHAN_PNETCDF_VAR_MOD-1) * sizeof(struct darshan_log_map));
+        memmove(&fd->mod_ver[DARSHAN_PNETCDF_VAR_MOD+1],
+            &fd->mod_ver[DARSHAN_PNETCDF_VAR_MOD],
+            (DARSHAN_MAX_MODS-DARSHAN_PNETCDF_VAR_MOD-1) * sizeof(uint32_t));
+        fd->mod_map[DARSHAN_PNETCDF_VAR_MOD].len =
+            fd->mod_map[DARSHAN_PNETCDF_VAR_MOD].off = 0;
+        fd->mod_ver[DARSHAN_PNETCDF_VAR_MOD] = 0;
+    }
```

Comment:
I think there's still an off-by-1 error for the labeling of partial data.

When I check `python -m pytest --pyargs darshan -n 10 -k "test_module_table"` the `actual` vs. `expected` partial data labels are retrieved for the wrong modules with an offset of 1 (third column, bottom right).

```
actual_mod_df:
                                                       0                                                  1
Log Filename                   partial_data_dxt.darshan                                                NaN
Runtime Library Version                           3.2.1                                                NaN
Log Format Version                                 3.21                                                NaN
POSIX (ver=4) Module Data                      0.14 KiB                                                NaN
MPI-IO (ver=3) Module Data                     0.12 KiB                                                NaN
STDIO (ver=2) Module Data                      0.06 KiB  ⚠ Module data incomplete due to runtime memory...
DXT_POSIX (ver=1) Module Data                574.73 KiB  ⚠ Module data incomplete due to runtime memory...
DXT_MPIIO (ver=2) Module Data                568.14 KiB                                                NaN
expected_df:
                                                       0                                                  1
Log Filename                   partial_data_dxt.darshan                                                NaN
Runtime Library Version                           3.2.1                                                NaN
Log Format Version                                 3.21                                                NaN
POSIX (ver=4) Module Data                      0.14 KiB                                                NaN
MPI-IO (ver=3) Module Data                     0.12 KiB                                                NaN
STDIO (ver=2) Module Data                      0.06 KiB                                                NaN
DXT_POSIX (ver=1) Module Data                574.73 KiB  ⚠ Module data incomplete due to runtime memory...
DXT_MPIIO (ver=2) Module Data                568.14 KiB  ⚠ Module data incomplete due to runtime memory...

```

Reply:
ooh, good catch. this might affect our HDF5 module, too, and just haven't caught it yet. i'll try to get a fix in here, and then open a PR with a fix for HDF5.

Reply:
Also, I've pushed a little Python stuff here--the CI should reflect what I'm currently facing now, I think.

Reply:
Looks like that's fixed via [8298213](https://github.com/darshan-hpc/darshan/pull/456/commits/8298213cb7c170af3ebe232e21a9844f2284c023)